### PR TITLE
Allow Buildship to separate runtime classpath based on source set information

### PR DESCRIPTION
**Pull request by donat**
This pull request contains all changes required for https://github.com/eclipse/buildship/issues/354:
- Source folders use separate output locations
- Each source folder and dependency define scope information

### CODE REVIEW

**Comment by donat** [comment]
I'm working on the performance failures...

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -28,6 +28,10 @@ Gradle now suppoprts version range in parent elements of POM, which was introduc
 IMPORTANT: if this is a patch release, ensure that a prominent link is included in the foreword to all releases of the same minor stream.
 Add-->
 
+### Eclipse plugin separates output folders
+
+The `eclipse` plugin now defines separate output directories for each source folders. Also, each source folder and dependency defines an additional `gradle_source_sets` classpath attribute. The attribute specifies to which sourceSet the entry belonged. Future [Buildship](http://eclipse.org/buildship) versions will use this information to separate source sets when launching Java applications within Eclipse.
```
Comment (oehme) [inline subprojects/docs/src/docs/release/notes.md]: ⭕️ folder

Comment (oehme) [inline subprojects/docs/src/docs/release/notes.md]: ❌ This section is outdated. It does not reflect the actual implementation. How about not going into this detail at all and just explaining what it does? I had a draft of that on [my branch](https://github.com/gradle/gradle/tree/oehme/eclipse/scopes).

Comment (donat) [reply subprojects/docs/src/docs/release/notes.md]: I added the text from your branch.

**Pull request merged**